설명 추가

jsh0128 opened the pull request:
기존 설명 추가 -
React의 최적화 방법, useMemo, 클래스형 컴포넌트, 함수형 컴포넌트의 차이, React 동작 원리 Virtual DOM 설명, es6 문법 중 생각나는대로 말해달라, DOCTYPE
새로 설명 추가 - 
함수형 프로그래밍, npm vs yarn, function, arrow function, (type, interface)차이, Memoization, 코드 스플리팅, JavaScript 자료형, React 최적화 방법 추가

jsh0128 committed:
설명 추가

React의 최적화 방법, useMemo, 클래스형 컴포넌트, 함수형 컴포넌트의 차이, React 동작 원리 Virtual DOM 설명, es6 문법 중 생각나는대로 말해달라, DOCTYPE에 설명추가
함수형 프로그래밍, npm vs yarn, function, arrow function, (type, interface)차이, Memoization, 코드 스플리팅, JavaScript 자료형, React 최적화 방법

jsh0128 committed:
내용 추가

Redux와 Redux-Saga, Redux-Thunk비교 추가, HTTP란 HTTP, HTTPS 차이, Restful API  설명, 웹 표준 설명, Reflow Repaint 설명 추가

Sh031224 commented on README.md:
```diff
@@ -174,6 +228,7 @@ React 부분
 16. useMemo
     - [] 안에 값이 바뀌었을 때만 다시 호출
     - useCallback과의 차이는 Callback은 함수를 반환, Memo는 값을 반환
+    - Hook이기 때문에 오직 함수형 컴포너틑 안에서만 사용 가능하다 - React.memo랑 차이점
```
여기 오타

Sh031224 commented on README.md:
```diff
@@ -183,3 +238,17 @@ React 부분
     - createRef을 함수형에서 작성 시 렌더링마다 ref가 초기화 된다.
+20. React의 최적화 방법
+    - useMemo를 사용
+    - React.PureComponent 사용 - 이전 props와 state를 얇게 비교하여 Component 렌더링 횟수 감소
```
React.memo도 비슷하지만 props만 비교 내용 추가하면 좋을듯!

jsh0128 replied:
나 이거 무슨 말인지 모르겠어

Sh031224 replied:
React.PureComponent 와 React.memo 둘이 내용 비슷하니까 같이 적으면 좋을듯 이야

Sh031224 commented on README.md:
```diff
@@ -118,14 +122,63 @@
+        - 순수함수 : 부수 효과가 없는 함수
+          함수 동일한 인자가 주어졌을 떄 항상 같은 값을 리턴하는 함수
+    - 가독성을 높이고 유지보수가 쉽다.
+47. Reflow Repaint 칭;
```
여기두 오타

Sh031224 commented on README.md:
```diff
@@ -118,14 +122,63 @@
+42. function, arrow function 차이
+    - function : this는 자신을 가장 마지막으로 품고 있는 scope로 항상 변한다.
+    - arrow function : 처음 바인딩 된 스코프 안에서 가리키는 this가 절대 변하지 않는다.
+44. npm vs yarn
```
위에 이미 내용있어서 거기에 추가해줘

jsh0128 replied:
음... 어디있는지 모르겠는데 
몇번인데??

Sh031224 replied:
npm vs yarn

Sh031224 commented on README.md:
```diff
@@ -118,14 +122,63 @@
     -  자신이 선언되었을때의 환경 밖에서 호출되어도 그 환경에 접근할 수 있는 함수
     - 전역 변수의 사용을 억제 하기위해
     - 정보를 은닉하기 위해 
+34. JavaScript 자료형
```
이것두 깊은복사 얕은복사 다루는 내용에 있어

jsh0128 committed:
Update README.md

jsh0128 committed:
오타 수정

jsh0128 committed:
설명 추가 및 정리

기존 설명 추가 -
React의 최적화 방법, useMemo, 클래스형 컴포넌트, 함수형 컴포넌트의 차이, React 동작 원리 Virtual DOM 설명, es6 문법 중 생각나는대로 말해달라, DOCTYPE
새로 설명 추가 -
함수형 프로그래밍, npm vs yarn, function, arrow function, (type, interface)차이, Memoization, 코드 스플리팅, JavaScript 자료형, React 최적화 방법 추가

Sh031224 commented on README.md:
```diff
@@ -183,3 +231,16 @@ React 부분
     - ref 객체 안에 current 속성이 실제 요소
     - createRef을 함수형에서 작성 시 렌더링마다 ref가 초기화 된다.
+20. React의 최적화 방법
+    - React.PureComponent, React.memo - 이전 props와 state를 얇게 비교하여 Component 렌더링 횟수 감소
```
`React.memo` 는 props만 비교!

jsh0128 committed:
Update README.md
